File: CCp/C++/Math/beginner/reachFast.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
  int T;
  cin >> T;
  while (T--)
  {

    int C, H, K;
    cin >> C >> H >> K;

    int diff = abs(C - H);
    if (diff == 0 ||  K < diff)
    {
      if (diff % K == 0)
      {
        cout << (diff / K) << endl;
      }
      else
      {
        cout << (diff / K) + 1 << endl;
      }
    }
    else
    {
      cout << "1\n";
    }
  }
}

File: CCp/TypeScript/src/Math/reachFast.ts

function reachFast ( c: number, h: number, k:number ) {
      
  let diff = Math.abs( c - h );
  console.log( (diff % k) ? ( Math.floor(diff / k) + 1 ) : Math.floor( diff / k));
}

reachFast(20, 10, 3);
reachFast(36, 36, 5);
reachFast(50, 4, 100);
reachFast(4, 30, 2);